frontend/catalyst/autograph/transformer.py

```diff
@@ -122,6 +122,9 @@ def run_autograph(fn):
     new_fn.ag_source_map = source_map
     new_fn.ag_unconverted = fn
 
+    if hasattr(fn, "execute_kwargs"):
+        new_fn.execute_kwargs = fn.execute_kwargs
+
```

Comment:
This code seems rather out of place here 🤔 What is your reasoning to add it here?

Reply:
``execute_kwargs`` contains ``mcm_config``. We have to pass it the new qnode so that the transform can happen.
I doubt that ``execute_kwargs`` might contain many other flags like ``mcm_config`` so I decide to copy it entirely.

Reply:
Right, but my main concern is that this happens in this function which doesn't care what its input is. Wouldn't this code be better suited where we copy the QNode?

Reply:
Yes, done.